### PR TITLE
Add a CI that test against vibe-d:tls

### PR DESCRIPTION
**Pull request by Geod24**
Because I'm lazy I don't want to have to do it myself every time :)

### CODE REVIEW

**Review by CyberShadow — APPROVED** [review]
:+1:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -113,3 +113,24 @@ jobs:
...
+      with:
+        repository: 'vibe-d/vibe.d'
+        # Use a fixed ref to avoid random breakage due to upstream
+        # The first release compatible with this CI is v0.9.5-beta.1,
+        # feel free to update on new releases (commits can also be used).
+        ref: 'v0.9.5-beta.1'
+
+    - name: 'Test with Vibe.d'
+      run: |
+        if [ ! -f $GITHUB_WORKSPACE/openssl/dub.sdl ]; then
```
Comment (CyberShadow) [inline .github/workflows/ci.yml:128]: I think the clone action sets `GITHUB_WORKSPACE` to the cloned repository.

**Comment by Geod24** [comment]
Right I misread the documentation.
```
    # Relative path under $GITHUB_WORKSPACE to place the repository
    path: ''
```
I assumed that the repository name was appended automatically.

**Comment by Geod24** [comment]
```
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/v0.9.5-beta.1*:refs/remotes/origin/v0.9.5-beta.1* +refs/tags/v0.9.5-beta.1*:refs/tags/v0.9.5-beta.1*
  /usr/lib/git-core/git-remote-https: /home/runner/work/openssl/openssl/openssl/install/lib/libcrypto.so.1.1: version `OPENSSL_1_1_1' not found (required by /lib/x86_64-linux-gnu/libssh.so.4)
  The process '/usr/bin/git' failed with exit code 128
```

Okay maybe removing OpenSSL from the system isn't *that* smart.

**Comment by Geod24** [comment]
Interesting that DMD works with 1.0.2 but not LDC :thinking:

**Comment by Geod24** [comment]
Ready to go. We just don't run the Vibe.d test for 1.0.2u

**Review by CyberShadow — COMMENTED** [review]
One thing comes to mind. Vibe.d, being a big project, is probably more sensitive to changes in compilers than just the bindings. Since we're pinning the Vibe.d version (which makes sense to me), I think it would make sense to also pin the compiler versions, instead of `dmd-latest` and `ldc-latest`.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -106,10 +136,27 @@ jobs:
...
         #$DC -run test.d
+
+    - name: 'Test with Vibe.d'
+      if: matrix.openssl.version != '1.0.2u'
+      run: |
+        if [ ! -f ${{ github.workspace }}/deimos-openssl/dub.sdl ]; then
+            ls -lR $GITHUB_WORKSPACE
+            echo "Error: Could not find dub.sdl at ${{ github.workspace }}/deimos-openssl/dub.sdl, see ls result above"
+            exit 1
+        fi
```
Comment (CyberShadow) [inline .github/workflows/ci.yml:158]: This can be removed now, I think.

**Pull request merged**